File: branches/Medusa/Source/Fantasy.Jobs/Tasks/WriteXmlTask.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Linq;
using System.Xml;

namespace Fantasy.Jobs.Tasks
{
    [Task("writeXml", Consts.XNamespaceURI, Description="Write xml to files")]
    public class WriteXmlTask : XmlTaskBase
    {
       

        [TaskMember("include", Flags = TaskMemberFlags.Input | TaskMemberFlags.Required, Description="The list of items to write to.")]
        public TaskItem[] Include { get; set; }

        [TaskMember("content", Flags = TaskMemberFlags.Input | TaskMemberFlags.Inline | TaskMemberFlags.Required, 
            Description="The XML content to write to.")]
        public XElement Content { get; set; }

        #region ITask Members

        public override bool Execute()
        {
            if (this.Include != null && this.Include.Length > 0)
            {
                

                XElement root = this.Content.Elements().First();
               

                foreach (TaskItem item in this.Include)
                {

                    string path = item["fullname"];

                    XmlWriter writer = XmlWriter.Create(path, this.XmlWriterSettings);
                    try
                    {
                        root.WriteTo(writer);
                    }
                    finally
                    {
                        writer.Close();
                    }
                }
            }

            return true;
        }

        #endregion
    }
}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/tasks/WriteXmlTask.java
package fantasy.jobs.tasks;

import fantasy.io.*;
import fantasy.servicemodel.*;
import fantasy.*;
import fantasy.jobs.*;
import fantasy.jobs.Consts;
public class WriteXmlTask extends XmlTaskBase
{


//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
	//[TaskMember("include", Flags = TaskMemberFlags.Input | TaskMemberFlags.Required, Description="The list of items to write to.")]
	private TaskItem[] privateInclude;
	public final TaskItem[] getInclude()
	{
		return privateInclude;
	}
	public final void setInclude(TaskItem[] value)
	{
		privateInclude = value;
	}

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
	//[TaskMember("content", Flags = TaskMemberFlags.Input | TaskMemberFlags.Inline | TaskMemberFlags.Required, Description="The XML content to write to.")]
	private XElement privateContent;
	public final XElement getContent()
	{
		return privateContent;
	}
	public final void setContent(XElement value)
	{
		privateContent = value;
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region ITask Members

	@Override
	public boolean Execute()
	{
		if (this.getInclude() != null && this.getInclude().length > 0)
		{


			XElement root = this.getContent().Elements().First();


			for (TaskItem item : this.getInclude())
			{

				String path = item.getItem("fullname");

				XmlWriter writer = XmlWriter.Create(path, this.getFormat());
				try
				{
					root.WriteTo(writer);
				}
				finally
				{
					writer.Close();
				}
			}
		}

		return true;
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion
}